src/App.jsx

```diff
@@ -14,33 +16,89 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+const SORT_FIELD_ALPHABET = 'alphabet';
+const SORT_FIELD_LENGTH = 'length';
+const SORT_FIELD_REVERSE = 'reverse';
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+export const App = () => {
+  const [visibleGoods, setVisibleGoods] = useState(goodsFromServer);
+  const [sortField, setSortField] = useState('');
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+  const sortByAlphabet = () => {
+    setVisibleGoods(
+      [...visibleGoods].sort((good1, good2) => good1.localeCompare(good2)),
+    );
+    setSortField(SORT_FIELD_ALPHABET);
+  };
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+  const sortByLength = () => {
+    setVisibleGoods(
+      [...visibleGoods].sort((good1, good2) => good1.length - good2.length),
+    );
+    setSortField(SORT_FIELD_LENGTH);
+  };
+
+  const reverse = () => {
+    setVisibleGoods([...visibleGoods].reverse());
+    setSortField(SORT_FIELD_REVERSE);
```

Comment:
combine all that logic with sorting and reversing in one function and remove state for goods, just write
`const visibleGoods = getVisibleGoods(goodsFromServer, sortValue, isReversed)`